### ISSUE TITLE
Mitmproxy is hanging when establishing connection ( libvirt + tls)

### ISSUE THREAD

**Issue by skotti** [mitmproxy/mitmproxy#3412]
Hello!
I am trying to put mitmproxy between client and server in KVM infrastructure. Client send rpc messages to server with the help of libvirt library. I don't actually know how to add mitmproxy certificate to libvirt trusted certificates, so I decided to use libvirt internal ca. It is my local ca, just generated with certtools. I launch mitmdump ( I set ssl-insecure ans I always get an error about SNI)

mitmdump --mode transparent --ssl-insecure --set confdir=/home/u/
Proxy server listening at http://*:8080
192.168.246.134:47104: clientconnect
::ffff:192.168.246.134:47104: Certificate verification error for None: unable to get local issuer certificate (errno: 20, depth: 0)
::ffff:192.168.246.134:47104: Ignoring server verification error, continuing with connection

After that mitmproxy is hanging, connection is not established and no messages are intercepted. What should be provided so that my CA could  work in mitmproxy?

home/u/my_project/bin/mitmdump --version
Mitmproxy: 4.0.4
Python:    3.6.7
OpenSSL:   OpenSSL 1.1.0i  14 Aug 2018
Platform:  Linux-4.15.0-42-generic-x86_64-with-Ubuntu-18.04-bionic

**Comment by skotti** [comment]
I want to update my question.  I signed now all my certificates with mitmproxy CA just for testing, Now the program is just hanging. I made a screenshot from wireshark for packets that are sent in two cases: the established tls connection without mitmproxy and not established with mitmproxy, maybe this will help.

![with_mitmproxy](https://user-images.githubusercontent.com/8792989/49697151-ee00ed00-fbc4-11e8-8030-70ef7db5c277.png)
![without_mitmproxy](https://user-images.githubusercontent.com/8792989/49697153-f0fbdd80-fbc4-11e8-95b9-38f04e0d911e.png)
